android/src/main/kotlin/com/adyen/checkout/flutter/components/base/ComponentSessionCallback.kt

```diff
@@ -41,7 +34,32 @@ abstract class ComponentSessionCallback<T : PaymentComponentState<*>>(
             ComponentCommunicationModel(
                 type = ComponentCommunicationType.RESULT,
                 componentId = componentId,
-                paymentResult = paymentResult
+                paymentResult =
+                    PaymentResultDTO(
+                        type = PaymentResultEnum.FINISHED,
+                        result = paymentResult
+                    )
+            )
+        componentFlutterApi.onComponentCommunication(model) {}
```

Comment:
Is this empty lambda `{}` intended?

Reply:
This lambda is the callback that pigeon uses for a back communication from the Flutter layer. The callback is being generated automatically and not optional. Therefore the empty lambda exists.